### PR TITLE
Cmake changes

### PR DESCRIPTION
**Pull request by AdithyaBaglody**
This patch now enabled the cmake build system for client-sdk.
To invoke the cmake cmd.

1. for default build
   $ cmake .
   $ make
2. For various build options.
   $ cmake -DDA=ecdsa -DKEX=ecdh -DPK_ENC=ecdsa .
   $ make

### CODE REVIEW

File: crypto/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  common/sdoKeyExchange.c
+  common/sdoAes.c
+  common/sdoHmac.c
+  common/sdoDevSign.c
+  common/sdoCryptoCommon.c
+  common/sdoDevAttest.c
+  )
+
+if (${KEX} MATCHES asym)
+        client_sdk_sources(common/sdokeyexchange_asym.c)
```
Comment (trbehera) [inline crypto/CMakeLists.txt:27]: Extra indent.

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  target_sources(linux-client PRIVATE app/blob.c)
+
+  if(${CRYPTO_HW} MATCHES true)
+    target_sources(linux-client PRIVATE app/se_provisioning.c)
+  endif()
+  target_include_directories(linux-client PRIVATE app/include)
+  target_link_libraries(linux-client client_sdk)
+
+
+client_sdk_ld_options(
```
[the remark this answers is not in the document]
Comment (AdithyaBaglody) [reply CMakeLists.txt]: yep that was miss. thanks

File: cDevice_Build.sh [hunk 3]
```diff
@@ -16,10 +16,10 @@ fi
 echo "For ecdsa_c_device *******"
 mkdir -p ecdsa256_c_device_bin
 mkdir -p ecdsa256_c_device_bin/blob_backup
-make DA=ecdsa256 pristine
-make BUILD=${BUILDTYPE} PK_ENC=ecdsa DA=ecdsa256 KEX=ecdh
+make pristine ||  cmake -DBUILD=${BUILDTYPE} -DPK_ENC=ecdsa -DDA=ecdsa256 -DKEX=ecdh
+make -j4
```
[the remark this answers is not in the document]
Comment (AdithyaBaglody) [reply cDevice_Build.sh]: fixed it with $(nproc)

File: cDevice_Build_NoUnitTest.sh [hunk 4]
```diff
@@ -31,10 +31,10 @@ cp -a ${WORKSPACE}/data/platform_iv.bin ${WORKSPACE}/ecdsa256_c_device_bin/blob_
 echo "For ecdsa_c_device for Supply chain tool*******"
 mkdir -p ecdsa256_c_sct_device_bin
 mkdir -p ecdsa256_c_sct_device_bin/blob_backup
-make DA=ecdsa256 pristine
-make BUILD=${BUILDTYPE} PK_ENC=ecdsa DA=ecdsa256 MANUFACTURER_TOOLKIT=true KEX=ecdh
+make pristine ||  cmake -DBUILD=${BUILDTYPE} -DPK_ENC=ecdsa -DDA=ecdsa256 -DMANUFACTURER_TOOLKIT=true -DKEX=ecdh
```
[the remark this answers is not in the document]
Comment (AdithyaBaglody) [reply cDevice_Build_NoUnitTest.sh]: i can make another PR after setting that as default . resolving this comment as of now

File: cmake/blob_path.cmake [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+# if absoulte is needed uncomment the following line
+#set(BLOB_PATH ${BASE_DIR})
+
+if(NOT (DEFINED BLOB_PATH))
+set(BLOB_PATH .)
+endif()
+
+if(TARGET_OS MATCHES linux)
+  
+  Client_sdk_compile_definitions(
```
[the remark this answers is not in the document]
Comment (AdithyaBaglody) [reply cmake/blob_path.cmake]: Fixed

File: cmake/blob_path.cmake [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+# SPDX-License-Identifier: Apache 2.0
+#
+
+
+# Note all blobs and data will be made relative.
+# if absoulte is needed uncomment the following line
+#set(BLOB_PATH ${BASE_DIR})
+
+if(NOT (DEFINED BLOB_PATH))
+set(BLOB_PATH .)
```
[the remark this answers is not in the document]
Comment (AdithyaBaglody) [reply cmake/blob_path.cmake]: fixed

File: cmake/blob_path.cmake [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+	if (${DA_FILE} MATCHES pem)
+	  client_sdk_compile_definitions(
+	    -DECDSA_PRIVKEY=\"${BLOB_PATH}/data/ecdsa384privkey.pem\")
+	else()
+	  client_sdk_compile_definitions(
+	    -DECDSA_PRIVKEY=\"${BLOB_PATH}/data/ecdsa384privkey.dat\")
+	endif()
+      endif()
+      client_sdk_compile_definitions(
+	-DSDO_CACERT=\"${BLOB_PATH}/data/cacert.bin\"
```
[the remark this answers is not in the document]
Comment (AdithyaBaglody) [reply cmake/blob_path.cmake:81]: fixed

File: crypto/CMakeLists.txt (hunk 1 above)
Comment (AdithyaBaglody) [reply crypto/CMakeLists.txt]: fixed

**Comment by AdithyaBaglody** [comment]
@trbehera please re-review.

File: cmake/blob_path.cmake [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+        -DSDO_PUBKEY=\"${BLOB_PATH}/data/test_pubkey.dat\"
+        -DSDO_SIGRL=\"${BLOB_PATH}/data/test_sigrl.dat\"
+        -DSDO_CRED_SECURE=\"${BLOB_PATH}/data/Secure.blob\"
+        -DSDO_CRED_MFG=\"${BLOB_PATH}/data/Mfg.blob\"
+        -DSDO_CRED_NORMAL=\"${BLOB_PATH}/data/Normal.blob\"
+        -DRAW_BLOB=\"${BLOB_PATH}/data/raw.blob\"
+        )
+    else() 				#Not unit tests
+      if (${DA} MATCHES ecdsa256)	#ecdsa 256 selected
+	if (${DA_FILE} MATCHES pem)
```
Comment (trbehera) [inline cmake/blob_path.cmake:64]: Another level of indent for this if … else .. block.

**Comment by trbehera** [comment]
The PR looks good. @AdithyaBaglody Could you please squash and repost?

**Comment by AdithyaBaglody** [comment]
> The PR looks good. @AdithyaBaglody Could you please squash and repost?

Unfortunately for a few reasons i will need to maintain this commit history.

**Comment by trbehera** [comment]
> The PR looks good. @AdithyaBaglody Could you please squash and repost?
> 
> Unfortunately for a few reasons i will need to maintain this commit history.

Ok.

**Pull request merged**